Tools/CMake/FindAMReXFFTW.cmake

```diff
@@ -7,45 +7,142 @@ Finds the FFTW library.
-    find_library(FFTW_LIBRARY NAMES "fftw3" HINTS ${FFTW_ROOT}/lib)
-    find_library(FFTWF_LIBRARY NAMES "fftw3f" HINTS ${FFTW_ROOT}/lib)
-    set(FFTW_LIBRARIES ${FFTW_LIBRARY} ${FFTWF_LIBRARY})
-endif()
+# Set the AMReX_FFTW_OMP=1 define on AMReX::FFTW if TRUE and print
+# a message
+#
+function(fftw_add_define HAS_FFTW_OMP_LIB)
+    if(HAS_FFTW_OMP_LIB)
+        message(STATUS "FFTW: Found OpenMP support")
+        target_compile_definitions(AMReX::FFTW INTERFACE AMReX_FFTW_OMP=1)
+    else()
+        message(STATUS "FFTW: Could NOT find OpenMP support")
+    endif()
+endfunction()
 
-include(FindPackageHandleStandardArgs)
+# Check if the found FFTW install location has an _omp library, e.g.,
+# libfftw3(f)_omp.(a|so) shipped and if yes, set the AMReX_FFTW_OMP=1 define.
+#
+function(fftw_check_omp library_paths fftw_precision_suffix)
+    find_library(HAS_FFTW_OMP_LIB fftw3${fftw_precision_suffix}_omp
+            PATHS ${library_paths}
+            # this is intentional, so we don't mix different FFTW installs
+            # and only check what is in the location hinted by the
+            # "library_paths" variable
+            NO_DEFAULT_PATH
+            NO_PACKAGE_ROOT_PATH
+            NO_CMAKE_PATH
+            NO_CMAKE_ENVIRONMENT_PATH
+            NO_SYSTEM_ENVIRONMENT_PATH
+            NO_CMAKE_SYSTEM_PATH
+            NO_CMAKE_FIND_ROOT_PATH
+    )
+    if(HAS_FFTW_OMP_LIB)
+        # the .pc files here forget to link the _omp.a/so files
+        # explicitly - we add those manually to avoid any trouble,
+        # e.g., in static builds.
+        target_link_libraries(AMReX::FFTW INTERFACE ${HAS_FFTW_OMP_LIB})
+    endif()
+
+    fftw_add_define("${HAS_FFTW_OMP_LIB}")
+endfunction()
 
-find_package_handle_standard_args(AMReXFFTW
-    REQUIRED_VARS FFTW_LIBRARIES FFTW_INCLUDES)
 
-mark_as_advanced(FFTW_LIBRARIES FFTW_INCLUDES)
+# Central FFTW3 Search ###############################################
+#
+# On Windows, try searching for FFTW3(f)Config.cmake files first
+#   Installed .pc files wrongly and unconditionally add -lm
+#   https://github.com/FFTW/fftw3/issues/236
+
+# On Linux & macOS, note Autotools install bug:
+#   https://github.com/FFTW/fftw3/issues/235
+# Thus, rely on .pc files
+
+set(AMReX_FFTW_SEARCH_VALUES PKGCONFIG CMAKE)
+set(AMReX_FFTW_SEARCH_DEFAULT PKGCONFIG)
+if(WIN32)
+    set(AMReX_FFTW_SEARCH_DEFAULT CMAKE)
+endif()
+set(AMReX_FFTW_SEARCH ${AMReX_FFTW_SEARCH_DEFAULT}
+        CACHE STRING "FFTW search method (PKGCONFIG/CMAKE)")
+set_property(CACHE AMReX_FFTW_SEARCH PROPERTY STRINGS ${AMReX_FFTW_SEARCH_VALUES})
+if(NOT AMReX_FFTW_SEARCH IN_LIST AMReX_FFTW_SEARCH_VALUES)
+    message(FATAL_ERROR "AMReX_FFTW_SEARCH (${AMReX_FFTW_SEARCH}) must be one of ${AMReX_FFTW_SEARCH_VALUES}")
+endif()
+mark_as_advanced(AMReX_FFTW_SEARCH)
+
+# floating point precision suffixes: float, double and quad precision
+if(AMReX_PRECISION STREQUAL "DOUBLE")
+    set(HFFTWp "")
+else()
+    set(HFFTWp "f")
+endif()
```

Comment:
We have to wrap everything that depends on `${HFFTWp}` into a function and call it twice if we want to make sure float and double were built into FFTW.